Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -0,0 +1,86 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('vsts-task-lib/task');
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
+
+// Creates the tool runner for executing SonarQube.
+export function applySonarQubeParams(toolRunner:ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?):ToolRunner {
+
+    toolRunner.arg('-Dsonar.host.url=' + sqHostUrl);
+    toolRunner.arg('-Dsonar.login=' + sqHostUsername);
+    toolRunner.arg('-Dsonar.password=' + sqHostPassword);
+    if (sqDbUrl) {
+        toolRunner.arg('-Dsonar.jdbc.url=' + sqDbUrl);
+    }
+    if (sqDbUsername) {
+        toolRunner.arg('-Dsonar.jdbc.username=' + sqDbUsername);
+    }
+    if (sqDbPassword) {
+        toolRunner.arg('-Dsonar.jdbc.password=' + sqDbPassword);
+    }
+
+    return toolRunner;
+}
+
+export class SonarQubeEndpoint {
+    constructor(public Url, public Username, public Password) {
```

Comment:
Yes - this is because Typescript will fill out the implied fields (with their appropriate modifiers as given in the constructor) during compile time.